python/cudf/cudf/core/column/column.py

```diff
@@ -385,25 +385,26 @@ def copy(self, deep=True):
                 children=self.base_children,
             )
 
-    def view(self, newcls, **kwargs):
-        """View the underlying column data differently using a subclass of
-        ColumnBase
+    def view(self, dtype):
+        """
+        View the data underlying a column as different dtype.
 
         Parameters
         ----------
-        newcls : ColumnBase
-            The logical view to be used
-        **kwargs :
-            Additional paramters for instantiating instance of *newcls*.
-            Valid keywords are valid parameters for ``newcls.__init__``.
-            Any omitted keywords will be defaulted to the corresponding
-            attributes in ``self``.
+        dtype : NumPy dtype, string
+            The dtype to view the data as
+        
         """
-        params = Column._replace_defaults(self)
-        params.update(kwargs)
-        if "mask" in kwargs and "null_count" not in kwargs:
-            del params["null_count"]
-        return newcls(**params)
+        if self.null_count > 0:
+            raise ValueError("Can not produce a view of a column with nulls")
```

Comment:
why is column with nulls not supported?

Reply:
I had some trouble reasoning about this in all cases. For instance, what if we have `cudf.Series([1, None], dtype='int32').view('int64')`?  I am not sure what the null mask would be in this situation. Although really it should allow cases where the dtypes are the same size, even with nulls.

Reply:
null mask would be exactly same (considering both of them have same offset). nullmask is 1 bit per element for any datatype in libcudf. 
I am not sure how offset is handled in python column view.

Reply:
I am not sure I follow. If we kept the same mask, we'd have a mask with length 2 referring to a series of length 1, right?

Reply:
Correct. It won't work for different length. it could still support columns of same return length! (it will be useful at few places)

Reply:
👍 I'll update this to require that `self.null_count == 0 and self.dtype.itemsize == dtype.itemsize`